syntax = "proto3";
package monitor.proto;

import "google/protobuf/empty.proto";
import "net_info.proto";
import "mem_info.proto";
import "cpu_stat.proto";
import "cpu_softirq.proto";
import "cpu_load.proto";
import "disk_info.proto";

message MonitorInfo{
  string name = 1;
  repeated SoftIrq soft_irq = 4;
  CpuLoad cpu_load = 5;
  repeated CpuStat cpu_stat = 6;
  MemInfo mem_info = 7;
  repeated NetInfo net_info = 8;
  repeated DiskInfo disk_info = 9;
}

service GrpcManager {
  rpc SetMonitorInfo(MonitorInfo) returns (google.protobuf.Empty) {
  }

  rpc GetMonitorInfo(google.protobuf.Empty) returns (MonitorInfo) {
  }
}
